ui/app/common/directives/pncScmValidator.js

```diff
@@ -54,7 +66,15 @@
           (exactHost ? exactHost + '(\\/[\\w\\.:~_-]+)+\\.git' : '[\\w\\.@:\\/~_-]+') + // repository
           //'\\.git' +                          // suffix
           '(?:\\/?|\\#[\\d\\w\\.\\-_]+?)$');  // parameters
-        return pattern.test(url);
+        if (pattern.test(url)) {
+          if (isRepoInternal(url) && isGitlabAndNotScpFormat(url)) {
+            return false;
+          } else {
+            return true;
+          }
+        } else {
+         return false;
+        }
```

Comment:
can be simplified by
`return pattern.test(url) && !(isRepoInternal(url) && isGitlabAndNotScpFormat(url));`

Reply:
True! Thanks for pointing this out!